pkg/drivers/kic/node/node.go

```diff
@@ -0,0 +1,233 @@
+	DefaultOci      = "docker"
+)
+
+// Node represents a handle to a kic node
+// This struct must be created by one of: CreateControlPlane
+type Node struct {
+	// must be one of docker container ID or name
+	name string
+	// cached node info etc.
+	cache *nodeCache
+	R     command.Runner // Runner
+}
+
+// WriteFile writes content to dest on the node
+func (n *Node) WriteFile(dest, content string, perm string) error {
+	// create destination directory
+	cmd := exec.Command("mkdir", "-p", filepath.Dir(dest))
+	rr, err := n.R.RunCmd(cmd)
+	if err != nil {
+		return errors.Wrapf(err, "failed to create directory %s cmd: %v output:%q", cmd.Args, dest, rr.Output())
+	}
+
+	cmd = exec.Command("cp", "/dev/stdin", dest)
+	cmd.Stdin = strings.NewReader(content)
+
+	if rr, err := n.R.RunCmd(cmd); err != nil {
+		return errors.Wrapf(err, "failed to run: cp /dev/stdin %s cmd: %v output:%q", dest, cmd.Args, rr.Output())
+	}
+
+	cmd = exec.Command("chmod", perm, dest)
+	_, err = n.R.RunCmd(cmd)
+	if err != nil {
+		return errors.Wrapf(err, "failed to run: chmod %s %s", perm, dest)
+	}
+	return nil
+}
+
+// IP returns the IP address of the node
+func (n *Node) IP() (ipv4 string, ipv6 string, err error) {
+	// use the cached version first
+	cachedIPv4, cachedIPv6 := n.cache.IP()
+	if cachedIPv4 != "" && cachedIPv6 != "" {
+		return cachedIPv4, cachedIPv6, nil
+	}
+	// retrieve the IP address of the node using docker inspect
+	lines, err := oci.Inspect(n.name, "{{range .NetworkSettings.Networks}}{{.IPAddress}},{{.GlobalIPv6Address}}{{end}}")
+	if err != nil {
+		return "", "", errors.Wrap(err, "failed to get container details")
+	}
+	if len(lines) != 1 {
+		return "", "", errors.Errorf("file should only be one line, got %d lines", len(lines))
+	}
+	ips := strings.Split(lines[0], ",")
+	if len(ips) != 2 {
+		return "", "", errors.Errorf("container addresses should have 2 values, got %d values", len(ips))
+	}
+	n.cache.set(func(cache *nodeCache) {
+		cache.ipv4 = ips[0]
+		cache.ipv6 = ips[1]
+	})
+	return ips[0], ips[1], nil
+}
+
+// LoadImageArchive loads an image from archive into the node
+func (n *Node) LoadImageArchive(image io.Reader) error {
+	cmd := exec.Command(
+		"ctr", "--namespace=k8s.io", "images", "import", "-",
+	)
+	cmd.Stdin = image
+	if _, err := n.R.RunCmd(cmd); err != nil {
+		return errors.Wrap(err, "failed to load image")
+	}
+	return nil
+}
+
+// Copy copies a local asset into the node
+func (n *Node) Copy(ociBinary string, asset assets.CopyableFile) error {
+	if err := oci.Copy(ociBinary, n.name, asset); err != nil {
+		return errors.Wrap(err, "failed to copy file/folder")
+	}
+
+	// TODO: medya verify add tests
+	cmd := exec.Command("chmod", asset.GetPermissions(), asset.GetTargetName())
+	if _, err := n.R.RunCmd(cmd); err != nil {
+		return errors.Wrap(err, "failed to chmod file permissions")
+	}
+	return nil
+}
+
+// Status gets status for node
+func (n *Node) Status() (state.State, error) {
+	return oci.Status(DefaultOci, n.name)
```

Comment:
Using DefaultOci here seems weird. What do you think about adding an oci attribute to the node so that it can do the right thing here instead of hardcoding Docker?